SM-59 Create endpoint to store a material for a reparation request - #59

JSTLWK opened the pull request:
# [WIP] Create endpoint to store a material for a reparation request - #59

**Changelog**
- Added store method for API.

JSTLWK committed:
SM-60 - Create endpoint to to retrieve a material for reparation request

- Added gitignore files
- Added packages for filters and sorting and json pagination.
- Created Index for all material reparations
- Edited env for testing; we should pick this up.

Commit:
style: fix code style

JSTLWK committed:
SM-59 - Create endpoint to store a material for a reparation request

- Added store method for API.

JSTLWK committed:
SM-59 - Create endpoint to store a material for a reparation request

- Added store method for API.

Commit:
style: fix code style

Commit:
style: fix code style

JSTLWK committed:
SM-59 - Create endpoint to store a material for a reparation request - Test creation

- Added test for create endpoint
- Added way of error handeling back to the client.

JSTLWK committed:
SM-59 - Create endpoint to store a material for a reparation request - Test creation

- Added test for create endpoint
- Added way of error handeling back to the client.

JSTLWK committed:
Merge branch 'feature/SM-59' of github.com:KA-Huis/space-management into feature/SM-59

Commit:
style: fix code style

Commit:
style: fix code style

JSTLWK committed:
SM-60 - Pull request changes

- reverted some committed changes.

cyrildewit commented on app/Http/Controllers/API/V1/ReparationRequestMaterialController.php:
```diff
@@ -34,4 +36,15 @@ public function index(Request $request): LengthAwarePaginator
+
+    public function store(ReparationRequestMaterialStoreRequest $reparationRequestMaterialStoreRequest): ReparationRequestMaterial
+    {
+        $reparationRequest = ReparationRequest::find((int) $reparationRequestMaterialStoreRequest->get('reparation_request_id'));
```
A policy is missing for this model and ability. Since we don't have a concrete solution fore this, we can specify the permission check logic later.

JSTLWK committed:
Merge branch 'feature/SM-60' into feature/SM-59

# Conflicts:
#	.php-cs-fixer.cache

Commit:
style: fix code style

cyrildewit commented on app/Http/Requests/API/FormRequest.php:
```diff
@@ -0,0 +1,42 @@
+use Illuminate\Http\JsonResponse;
+use Illuminate\Validation\ValidationException;
+
+abstract class FormRequest extends LaravelFormRequest
```
This solution is not needed as discussed. If you set the `Accept` header to `application/json` it will automatically return the form validation errors in a JSON format.

cyrildewit commented on app/Http/Requests/API/V1/ReparationRequestMaterialStoreRequest.php:
```diff
@@ -0,0 +1,24 @@
+        return [
+            'name' => ['required', 'string', 'max:255'],
+            'is_mandatory' => ['required', 'boolean'],
+            'reparation_request_id' => ['required', 'exists:App\Models\ReparationRequest,id'],
```
I prefer to use `Rule::exists()` since the API is then more explicit. Right know, I will have to read the documentation. It's also more OOP.

JSTLWK committed:
Merge branch 'feature/SM-59' of github.com:KA-Huis/space-management into feature/SM-59

# Conflicts:
#	.php-cs-fixer.cache
#	composer.json
#	composer.lock
#	routes/api.php

Commit:
style: fix code style

JSTLWK committed:
Merge branch 'feature/SM-60' into feature/SM-59

# Conflicts:
#	.php-cs-fixer.cache
#	composer.lock
#	routes/api.php

Commit:
style: fix code style

JSTLWK committed:
SM-59: Create endpoint to store a material for a reparation request - Pull request feedback

This is a combination of both commented feedback and feedback we have discussed from the other pull request.

Commit:
style: fix code style

cyrildewit commented on app/API/V1/Http/Controllers/ReparationRequestMaterialController.php:
```diff
@@ -33,4 +35,15 @@ public function index(): ReparationRequestMaterialCollection
+        $reparationRequestMaterial->reparationRequest()->associate($reparationRequest);
+        $reparationRequestMaterial->save();
+
+        return $reparationRequestMaterial;
```
This should return the resource.

JSTLWK committed:
Merge branch 'acceptance' into feature/SM-59

# Conflicts:
#	.php-cs-fixer.cache

Commit:
style: fix code style

cyrildewit reviewed (changes requested):
See comments.

JSTLWK committed:
SM-59: Create endpoint to store a material for a reparation request - Pull request feedback

This is a combination of both commented feedback and feedback we have discussed from the other pull request.

Commit:
style: fix code style

cyrildewit reviewed (approved):
LGTM
